tests/credential_plugins_test.py

```diff
@@ -134,6 +135,68 @@ def test_hashivault_handle_auth_not_enough_args() -> None:
         hashivault.handle_auth()
 
 
+@pytest.mark.parametrize(
+    'explicit_creds',
+    (
+        {
+            'access_key': 'my_access_key',
+            'secret_key': 'my_secret_key',
+        },
+        {},
+    ),
+    ids=('with-creds-args', 'with-env-creds'),
+)
+@pytest.mark.parametrize(
+    (
+        'identifier_key',
+        'expected',
+    ),
+    (
+        (None, 'the_access_token'),
+        ('access_key', 'the_access_key'),
+        ('secret_key', 'the_secret_key'),
+    ),
+    ids=(
+        'access-token',
+        'access-key',
+        'secret-key',
+    ),
+)
+def test_aws_assumerole_identifier(
+    monkeypatch: pytest.MonkeyPatch,
+    explicit_creds: dict[str, str], identifier_key: str | None, expected: str,
+) -> None:
+    """Test that the aws_assumerole_backend function call returns a token given
+    the access_key and secret_key."""
+
+    def mock_getcreds(
+            access_key: str | None,
+            secret_key: str | None,
+            role_arn: str | None,
+            external_id: int,
+    ) -> dict:
+        return {
+            'access_key': 'the_access_key',
+            'secret_key': 'the_secret_key',
+            'access_token': 'the_access_token',
+            'Expiration': datetime.datetime.today() + datetime.timedelta(days=1),
```

Comment:
According to https://youtype.github.io/boto3_stubs_docs/mypy_boto3_sts/type_defs/#credentialstypedef, this should be
```suggestion
            'AccessKeyId': 'the_access_key',
            'SecretAccessKey': 'the_secret_key',
            'SessionToken': 'the_access_token',
            'Expiration': datetime.datetime.today() + datetime.timedelta(days=1),
```

@thedoubl3j do you know why the example return value diverges in tests so much?

Reply:
(this is the reason I don't like mocking, as it just tests the stubs instead of the important bits of code)

Reply:
I have no idea @webknjaz all of this just pulled from the AWX PR that we had from the external contributor so reasoning etc on the decisions made I have no clue.

Reply:
HI @webknjaz and @thedoubl3j I'm the original submitter of the PR, and you can pretty much explain any of my decisions as mistakes by a very new Python developer.

In this particular example, you're right, the actual call to boto3 should return an 'AccessKeyId', 'SecretAccessKey', 'SessionToken' 'Expiration' dict, but the mock was poorly written. Sorry about that.

I also agree that the mocking was a poor choice here, but given that the actual requirement to test that code is a valid AWS account with IAM set up, I wasn't sure of a better way (at least in the AWX project tests which didn't seem to have any of that capability) - if there is I'd love to know about it for any future work I do!

Reply:
Also please feel free to ping me if there's anything else I can help with on this PR.

Reply:
@derekwaters thanks! We'll keep this in mind. Having a full e2e test would be hard to achieve here and is to worth the effort. Normally, for such testing I'd go for something like a stub server to produce better coverage and test realistic behavior.
It's probably difficult to emulate in this case so the next best thing is to mock the client so that everything else runs as usual. Ideally, we should only mock things that are on the boundary between our code and third party, or something in the environment (like freezing time). These are nor particularly connected to pythonic vs. not. Unfortunately, oftentimes pythonistas don't know they can follow battle-tested architectural principles that are well-known in other ecosystems, not being exposed to their existence mostly. For example, I really like the dependency injection principle, which allows writing testable code and I believe we should design it to be testable from the very beginning, instead of testing as an afterthought and having to fight the previous decisions.